### PR TITLE
Adding recipe for conda-auth

### PR DESCRIPTION
**Pull request by travishathaway**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by jezdez** [comment]
Yep, happy to be a maintainer of this new package

File: recipes/conda-auth/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  url: https://github.com/conda-incubator/{{ name }}/archive/refs/tags/{{ version}}.tar.gz
+  sha256: d909a24adc9856119e80ac87658ee380d007d96f022ccb27a11d6acbce72234d
+
+build:
+  noarch: python
+  script: {{ PYTHON }} -m pip install . -vv
+  number: 0
+
+requirements:
+  host:
```
Comment (synapticarbors) [inline recipes/conda-auth/meta.yaml:18]: The `[build-system]` in `pyproject.toml` says that `setuptools-scm` is required.

Comment (travishathaway) [reply recipes/conda-auth/meta.yaml:18]: Will add it.

**Comment by synapticarbors** [comment]
Something weird is going on with the config of the upstream package. When I download the source from github and then run `python -m build -s`, then inspect what is in the `sdist`, I see the following:

```zsh
$ exa --tree
.
├── conda_auth
│  ├── conda_auth.egg-info
│  │  ├── dependency_links.txt
│  │  ├── entry_points.txt
│  │  ├── PKG-INFO
│  │  ├── requires.txt
│  │  ├── SOURCES.txt
│  │  └── top_level.txt
│  └── handlers
│     ├── __init__.py
│     ├── base.py
│     ├── basic_auth.py
│     └── oauth2.py
├── LICENSE
├── PKG-INFO
├── pyproject.toml
├── README.md
├── setup.cfg
└── tests
```

It looks like it's not actually picking up any of the files (including the `__init__.py` in the source directory.

**Comment by travishathaway** [comment]
Hi @synapticarbors,

Thanks for your help on this!

I believe I have fixed the problem. It had to do with a missconfigured `pyproject.toml` file. I've since deleted and recreated a new release on my GitHub project.

**Pull request merged**